src/util/transaction/transactionUtils.ts

```diff
@@ -51,17 +52,24 @@ export async function getAppCallInnerAssetData(
   txGroup: SignerTransaction[]
 ): Promise<AssetWithIdAndAmount[] | undefined> {
   const innerTxns = await getAppCallInnerTxns(client, txGroup);
+  const accountAddress = extractAccountAddressFromTxGroup(txGroup);
```

Comment:
let's be more descriptive as it is a bit difficult to follow the logic
```suggestion
  // Get the account address that will receive the assets after a certain operation is successful.
  // This equals to the address of the account that signs the txGroup.
  const receivingAccountAddress = extractAccountAddressFromTxGroup(txGroup);
```

Using the first txn in txGroup to extract the address of the signer account is still a bit worrisome for me. It seems to work for our cases (swap, add liquidity, etc.) since txGroup only includes txns that would be signed by the user's account (the account that will receive the goods after the operation). I think we can go on by extracting it from the first txn even though there is slight chance for an error with third party integrations. But I guess we could assist third party integrations that might encounter error with regard to this.

`getAppCallInnerTxns` also assumes that the only app call in txGroup is our app call. We could also further fix that by checking not only the type of the txn to be `appl` but by also checking if the `appIndex` is the expected one. This way it makes sure the innerTxns we extract for it are the ones that belong to the Tinyman application(s).

@jamcry can you also go over this?